Source/CarthageKit/Git.swift

```diff
@@ -57,6 +57,30 @@ public struct GitURL: Equatable {
 		}
 	}
 
+	/// Amount of time before a git repository is fetched again. Defaults to 1 minute
+	public static var fetchCacheInterval: NSTimeInterval = 60.0
+
+	/// Cache of fetch times used to reduce redundant fetches
+	private static var lastFetchTimes: [GitURL : NSTimeInterval] = [:]
+
+	static func clearFetchTimes() {
+		lastFetchTimes.removeAll()
+	}
+
+	var needsFetch: Bool {
+		guard let lastFetch = GitURL.lastFetchTimes[self] else {
+			return true
+		}
+
+		let difference = NSDate().timeIntervalSince1970 - lastFetch
+
+		return !(0...GitURL.fetchCacheInterval).contains(difference)
```

Comment:
This is extremely inefficient. Looking over an array of N elements just to see if `difference` is contained in the interval.